service/controller/appvalue/cache.go

```diff
@@ -0,0 +1,215 @@
+	}
+
+	if key.AppCatalogConfigMapName(*appCatalog) != "" {
+		configMaps = append(configMaps, index{
+			Name:      key.AppCatalogConfigMapName(*appCatalog),
+			Namespace: key.AppCatalogConfigMapNamespace(*appCatalog),
+		})
+	}
+
+	if key.AppConfigMapName(cr) != "" {
+		configMaps = append(configMaps, index{
+			Name:      key.AppConfigMapName(cr),
+			Namespace: key.AppConfigMapNamespace(cr),
+		})
+	}
+
+	if key.UserConfigMapName(cr) != "" {
+		configMaps = append(configMaps, index{
+			Name:      key.UserConfigMapName(cr),
+			Namespace: key.UserConfigMapNamespace(cr),
+		})
+	}
+
+	switch eventType {
+	case watch.Added, watch.Modified:
+		for _, configMap := range configMaps {
+			// First, put the watchUpdate label
+			err := c.addLabel(ctx, configMap)
+			if err != nil {
+				c.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("failed to add a label into configmap %#q in namespace %#q", configMap.Name, configMap.Namespace), "stack", fmt.Sprintf("%#v", err))
+				continue
+			}
+
+			v, ok := c.configMapToApps.Load(configMap)
+			if ok {
+				storedIndex, ok := v.(map[index]bool)
+				if !ok {
+					return microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", []index{}, v)
+				}
+
+				storedIndex[appIndex] = true
+				c.configMapToApps.Store(configMap, storedIndex)
+			} else {
+				m := map[index]bool{
+					appIndex: true,
+				}
+				c.configMapToApps.Store(configMap, m)
+			}
+		}
+
+	case watch.Deleted:
+		for _, configMap := range configMaps {
+			v, ok := c.configMapToApps.Load(configMap)
+			if ok {
+				storedIndex, ok := v.(map[index]bool)
+				if !ok {
+					return microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", []index{}, v)
+				}
+
+				delete(storedIndex, appIndex)
+				if len(storedIndex) == 0 {
+					err := c.removeLabel(ctx, configMap)
+					if err != nil {
+						c.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("failed to remove a label into configmap %#q in namespace %#q", configMap.Name, configMap.Namespace), "stack", fmt.Sprintf("%#v", err))
+						continue
+					}
+
+					c.configMapToApps.Delete(configMap)
+				} else {
+					c.configMapToApps.Store(configMap, storedIndex)
+				}
+			}
+		}
+
+	default:
+		c.logger.Log("debug", fmt.Sprintf("event %#q for app %#q is not supported", eventType, cr.Name))
+	}
+
+	return nil
+}
+
+func (c *AppValue) addLabel(ctx context.Context, cm index) error {
+	currentCM, err := c.k8sClient.K8sClient().CoreV1().ConfigMaps(cm.Namespace).Get(ctx, cm.Name, metav1.GetOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if c.selector.Matches(labels.Set(currentCM.Labels)) {
+		// no-op
+		return nil
+	}
+
+	patches := []patch{}
+
+	if len(currentCM.GetLabels()) == 0 {
+		patches = append(patches, patch{
+			Op:    "add",
+			Path:  "/metadata/labels",
+			Value: map[string]string{},
+		})
+	}
+
+	patches = append(patches, patch{
+		Op:    "add",
+		Path:  fmt.Sprintf("/metadata/labels/%s", replaceToEscape(label.AppOperatorVersion)),
```

Comment:
put the same version label as app CRs to make configmaps reconciled by the same app-operator.